packages/server/src/fhir/accesspolicy.ts

```diff
@@ -32,9 +32,12 @@ export type PopulatedAccessPolicy = AccessPolicy & { resource: AccessPolicyResou
  * @returns A repository configured for the login details.
  */
 export async function getRepoForLogin(authState: AuthState, extendedMode?: boolean): Promise<Repository> {
-  const { project, login, membership, onBehalfOfMembership } = authState;
+  const { login, membership: realMembership, onBehalfOfMembership } = authState;
+  const membership = onBehalfOfMembership ?? realMembership;
+  const systemRepo = getSystemRepo();
   const accessPolicy = await getAccessPolicyForLogin(authState);
 
+  const project = await systemRepo.readReference(membership.project);
```

Comment:
Is there a reason that we need to re-read the project here?  It's already available in `authState`.

Reply:
Yes, because the Project in `authState` is the "real" one (i.e. the Super Admin project) in this case